pkg/cleaner/aws/aws.go

```diff
@@ -52,13 +60,12 @@ func (a *Cleaner) Clean() error {
 	}
 
 	for _, f := range cleaners {
+		a.logger.Log("level", "debug", "message", fmt.Sprintf("running cleaner %s", getFunctionName(f)))
 		err := f()
 		if err != nil {
-			return microerror.Mask(err)
+			a.logger.Log("level", "error", "message", "error in cleaner %s: %#v", getFunctionName(f), err)
```

Comment:
I would actually return the error here, and log it outside the function.

Reply:
logging the error is a good point, but I think the function should have a chance to fail if anything goes wrong. Otherwise I am afraid we are just "hiding" those error and won't get a chance to fix them when they appear.

So how about collecting errors and returning them in the end after the loop has executed ?

Reply:
@TheoBrigitte So both logging error details and returning an array of errors?